modules/xmpp/XmppConnection.js

```diff
@@ -432,8 +432,33 @@ export default class XmppConnection extends Listenable {
         const { shard, websocketKeepAliveUrl } = this._options;
         const url = websocketKeepAliveUrl ? websocketKeepAliveUrl
             : this.service.replace('wss://', 'https://').replace('ws://', 'http://');
-
+        
+        // #bloomberg #shard @rpang27 use prodody url to get shard
+        const getShardUrl = this.service.replace('wss://', 'https://').replace('ws://', 'http://').replace('xmpp-websocket', 'shard');
```

Comment:
Shall we reuse `url` that is already resolved?

```js
const shardUrl = url.replace('xmpp-websocket', 'shard');
```

Reply:
In case websocketKeepAliveUrl is set but we don't want to use that